build-support/travis/travis.yml.mustache

```diff
@@ -105,16 +109,20 @@ pants_run_cache_config: &pants_run_cache_config
 # we directly clone the Pyenv repo.
 
 pyenv_setup: &pyenv_setup >
-  if [ ! -d "${PYENV_BIN}" ]; then
-  rm -rf "${PYENV_ROOT}";
-  git clone https://github.com/pyenv/pyenv "${PYENV_ROOT}";
+  if [[ ! -x "${PYENV_BIN}" ]]; then
```

Comment:
Is the travis cache strictly for earlier iterations of the same shard? Or is it populated by the bootstrap shards, then used by the other ones?

Reply:
Because we assign a unique `CACHE_NAME` to each shard, each shard gets its own cache.

The reason for the `! -x ` check is that Travis will always create the folder `${PYENV_ROOT}` no matter what at the beginning, even if the cache is empty. This would lead to a git clone failure.